Sources/SwiftSyntaxBuilder/Buildables.swift.gyb

```diff
@@ -0,0 +1,172 @@
+%{
+  from gyb_syntax_support import *
+  from gyb_syntax_support.kinds import lowercase_first_word
+  from gyb_syntax_support.kinds import SYNTAX_BASE_KINDS
+  # -*- mode: Swift -*-
+  # Ignore the following admonition it applies to the resulting .swift file only
+
+  def syntax_buildable_child_type(syntax_kind, is_token=False, is_optional=False):
+    if syntax_kind in SYNTAX_BASE_KINDS:
+      buildable_type = syntax_kind + 'Buildable'
+    elif not is_token:
+      buildable_type = syntax_kind
+    elif 'List' in syntax_kind:
```

Comment:
This is needed to add correct type (herre)[https://github.com/apple/swift-syntax/pull/268/files#diff-685068f90f6719a81556c55a8d5cd4e0b2d4f3fe395a24a21af3dbcbb1d46165R1539] etc.

Reply:
I have just tried this, and ended up with something that we wanted.

Look at the example below:
```swift
public struct AvailabilityVersionRestriction: SyntaxBuildable {
  let platform: TokenSyntax
  let version: VersionTupleSyntax?

  public init(
    platform: TokenSyntax,
    version: VersionTupleSyntax?
  ) {
    self.platform = platform
    self.version = version
  }

  ...
}
```

But we want 

```swift
public struct AvailabilityVersionRestriction: SyntaxBuildable {
  let platform: TokenSyntax
  let version: VersionTuple?

  public init(
    platform: TokenSyntax,
    version: VersionTuple?
  ) {
    self.platform = platform
    self.version = version
  }

  ...
}
```